File: core/src/main/java/exchange/utils/CurrencyUtils.java
package exchange.utils;

import static exchange.app.internal.api.model.Direction.BUY;

import exchange.app.internal.api.model.Direction;
import exchange.app.internal.api.model.Pair;

public class CurrencyUtils {

  public static String pairToCurrency(final Pair currencyChange, final Direction direction) {
    if (BUY.equals(direction)) {
      return currencyChange.name().split("_")[1];
    } else {
      return currencyChange.name().split("_")[0];
    }
  }
}

File: frontend/projects/shared-modules/src/lib/utils/currency-utils.ts
import { PairUtils } from './pair-utils';
import { UserTicket } from '../api/model/userTicket';

export class CurrencyUtils {
  public static ticketToCurrency(ticket: UserTicket | null): string {
    if (ticket == null) {
      return '';
    }
    if (ticket.direction == 'BUY') {
      return PairUtils.getQuoteCurrency(ticket.pair);
    } else {
      return PairUtils.getBaseCurrency(ticket.pair);
    }
  }
}
